common/nym-lp/src/state_machine.rs

```diff
@@ -118,44 +129,76 @@ impl LpStateMachine {
+    /// * `remote_ed25519_key` - Peer's Ed25519 public key for PSQ authentication and X25519 derivation
+    /// * `salt` - Fresh salt for PSK derivation (must be unique per session)
+    ///
+    /// # Errors
     ///
-    /// Requires the local *full* keypair to get the public key for lp_id calculation.
+    /// Returns `LpError::Ed25519RecoveryError` if Ed25519→X25519 conversion fails for the remote key.
+    /// Local private key conversion cannot fail.
     pub fn new(
         is_initiator: bool,
-        local_keypair: &Keypair, // Use Keypair
-        remote_public_key: &PublicKey,
-        psk: &[u8],
-        // session_manager: Arc<SessionManager> // Optional
+        local_ed25519_keypair: (&ed25519::PrivateKey, &ed25519::PublicKey),
+        remote_ed25519_key: &ed25519::PublicKey,
+        salt: &[u8; 32],
     ) -> Result<Self, LpError> {
-        // Calculate the shared lp_id// Calculate the shared lp_id
-        let lp_id = make_lp_id(local_keypair.public_key(), remote_public_key);
+        // AIDEV-NOTE: Ed25519→X25519 conversion for API simplification
+        // We use standard RFC 7748 conversion to derive X25519 keys from Ed25519 identity keys.
+        // This allows callers to provide only Ed25519 keys (which they already have for signing/identity)
+        // without needing to manage separate X25519 keypairs.
+        //
+        // Security: Ed25519→X25519 conversion is cryptographically sound (RFC 7748).
+        // The derived X25519 keys are used for:
+        // - Noise protocol ephemeral DH
+        // - PSQ ECDH baseline security (pre-quantum)
+        // - lp_id calculation (session identifier)
+
+        // Convert Ed25519 keys to X25519 for Noise protocol
+        let local_x25519_private = local_ed25519_keypair.0.to_x25519();
+        let local_x25519_public = local_ed25519_keypair
+            .1
+            .to_x25519()
+            .map_err(LpError::Ed25519RecoveryError)?;
+
+        let remote_x25519_public = remote_ed25519_key
+            .to_x25519()
+            .map_err(LpError::Ed25519RecoveryError)?;
+
+        // Convert nym_crypto X25519 types to nym_lp keypair types
+        let lp_private = LpPrivateKey::from_bytes(local_x25519_private.as_bytes());
+        let lp_public = LpPublicKey::from_bytes(local_x25519_public.as_bytes())?;
+        let lp_remote_public = LpPublicKey::from_bytes(remote_x25519_public.as_bytes())?;
+
+        // Create X25519 keypair for Noise and lp_id calculation
+        let local_x25519_keypair = Keypair::from_keys(lp_private, lp_public);
 
-        let local_private_key = local_keypair.private_key().to_bytes();
-        let remote_public_key = remote_public_key.as_bytes();
+        // Calculate the shared lp_id using derived X25519 keys
+        let lp_id = make_lp_id(local_x25519_keypair.public_key(), &lp_remote_public);
 
-        // Create the session immediately
+        // Create the session with both Ed25519 (for PSQ auth) and derived X25519 keys (for Noise)
```

Comment:
this looks like a big red flag to me: using the same underlying keys for two completely different purposes. @aniampio / @georgio mind taking a look at this?